src/components/VButton.vue

```diff
@@ -0,0 +1,231 @@
+<template>
+  <Component
+    :is="as"
+    ref="node"
+    :type="typeValue"
+    :role="role"
+    v-bind="filteredProps"
+    :class="[
+      $style.button,
+      $style[variant],
+      pressed && $style[`${variant}-pressed`],
+      $style[`size-${size}`],
+    ]"
+    :aria-pressed="pressed"
+    :disabled="disabledValue"
+    :tabindex="tabIndex"
+    v-on="$listeners"
+  >
+    <slot />
+  </Component>
+</template>
+
+<script>
+import { defineComponent, ref, watch, toRefs } from '@nuxtjs/composition-api'
+import { isButton } from 'reakit-utils'
+import { getTabIndex } from '~/utils/a11y/get-tab-index'
+import { isNativeTabbable } from '~/utils/a11y/is-native-tabbale'
+import { supportsDisabledAttribute } from '~/utils/a11y/supports-disabled-attribute'
+
+/**
+ * A button component that behaves just like a regular HTML `button` element
+ * aside from pre-applied styles based on the passed in variant.
+ *
+ * All props available for the basic `button` component are available here as
+ * well, including an `as` prop which allows for component polymorphism. The
+ * most common use case for this prop is to turn the `VButton` component into
+ * an `anchor` element, so that you can render a link instead of a `button`.
+ *
+ * The accessibility helpers on this component are critical and are completely
+ * adapted from Reakit's Button, Clickable, and Tabbable component implementations.
+ */
+const VButton = defineComponent({
+  name: 'VButton',
+  props: {
+    as: {
```

Comment:
Would be nice to add docstrings for the props as they show up in the Storybook args table.